docs/references/expo/custom-auth-flows/expo-impersonation.mdx

```diff
@@ -0,0 +1,491 @@
+
+```tsx
+// useImpersonatedUser
+
+function useImpersonatedUser(
+  actorSub: string,
+  setImpersonator: React.Dispatch<React.SetStateAction<string | UserDataJSON>>
+) {
+  React.useEffect(() => {
+    const getImpersonatedUser = async () => {
+      const res = await fetch(`/getImpersonatedUser`, {
+        method: "POST",
+        body: JSON.stringify({
+          impersonator_id: actorSub,
+        }),
+      });
+
+      const data = await res.json();
+
+      setImpersonator(data);
+
+      getImpersonatedUser();
+    };
+  }, [actorSub]);
+}
+```
+
+With our hook setup we now can create our API route that will call Clerk’s retrieve user endpoint, `/users`, and get back the Impersonated user’s full `User` object.
+
+```tsx
+// getImpersonatedUser+api.tsx
+
+export async function POST(request: Request) {
+  const body: { impersonator_id: string } = await request.json();
+
+  const { impersonator_id } = body;
+  const res = await fetch(`https://api.clerk.com/v1/users/${impersonator_id}`, {
+    headers: {
+      Authorization: `Bearer ${process.env.CLERK_SECRET_KEY}`,
+    },
+  });
+  const data = await res.json();
+
+  return Response.json(data);
+}
+```
+
+Now that we have the functionality for generating an impersonation session & getting their user data, we have a few more functions to complete that will get the `ticket` from the URL generated by our `useImpersonation` hook, pass that to our `signIn` function to create a new impersonated session, and allow for logout of the impersonated or the impersonator. 
+
+## Generate an impersonation session
+
+Let’s create our functions that work together to generate a new impersonate session, `extractTicket` and `impersonateUser`
+
+```tsx
+  // use this function to get the ticket ID from the response of our generateActorToken API
+  function extractTicketValue(input: string): string | undefined {
+    const index = input.indexOf("ticket=");
+    if (index !== -1) {
+      return input.slice(index + 7);
+    }
+    return undefined;
+  }
+	
+	// passing the ticket ID to our `signIn.create` function will create a new impersonated session that we can set to the active session
+  async function impersonateUser() {
+    if (!isLoaded) return;
+
+    if (typeof actorRes?.url === "string") {
+      const ticket = extractTicketValue(actorRes.url);
+
+      if (ticket) {
+        try {
+          const { createdSessionId } = await signIn.create({
+            strategy: "ticket",
+            ticket,
+          });
+
+          await setActive({ session: createdSessionId });
+          await user?.reload();
+
+          router.replace("/dashboard");
+        } catch (err) {
+          console.log(err);
+        }
+      }
+    }
+  }
+```
+
+Finally, we have our functionality used to signout. Usually this can be as simple as calling Clerk’s `signOut` function, but since we’re handling multiple sessions we’ll need to add some checks to make sure we’re signing out of the right session. 
+
+```tsx
```

Comment:
````suggestion

### Create a hook for signing out

Finally, create a hook for signing out. Usually this can be as simple as calling Clerk’s [`signOut`](https://clerk.com/docs/pr/1038/references/javascript/clerk/clerk#sign-out) function, but since to handling multiple sessions you must add some checks to ensure you're signing out of the right session. 

```tsx filename="app/(dashboard)/index.tsx"
````